src/foundry/common/documents/table-result.mjs.d.ts

```diff
@@ -0,0 +1,153 @@
+   * @param context - Construction context options
+   */
+  constructor(data: BaseTableResult.ConstructorData, context?: DocumentConstructionContext);
+
+  static override metadata: Readonly<BaseTableResult.Metadata>;
+
+  static override defineSchema(): BaseTableResult.Schema;
+
+  override testUserPermission(
+    user: foundry.documents.BaseUser,
+    permission: keyof typeof CONST.DOCUMENT_OWNERSHIP_LEVELS | CONST.DOCUMENT_OWNERSHIP_LEVELS,
+    {
+      exact
+    }?: {
+      /**
+       * Require the exact permission level requested?
+       * @defaultValue `false`
+       */
+      exact?: boolean;
+    }
+  ): boolean;
+
+  static override migrateData(source: object): object;
+
+  static override shimData(
+    data: object,
+    {
+      embedded
+    }?: {
+      /**
+       * Apply shims to embedded models?
+       * @defaultValue `true`
+       */
+      embedded?: boolean;
+    }
+  ): object;
+}
+export default BaseTableResult;
+
+declare namespace BaseTableResult {
+  type Metadata = Merge<
+    DocumentMetadata,
+    {
+      name: "TableResult";
+      collection: "results";
+      label: "DOCUMENT.TableResult";
+      labelPlural: "DOCUMENT.TableResults";
+      coreTypes: ["0", "1", "2"];
+      permissions: {
+        update: (user: documents.BaseUser, doc: Document.Any, data: UpdateData) => boolean;
+      };
+    }
+  >;
+
+  type SchemaField = fields.SchemaField<Schema>;
+  type ConstructorData = UpdateData;
+  type UpdateData = fields.SchemaField.InnerAssignmentType<Schema>;
+  type Properties = fields.SchemaField.InnerInitializedType<Schema>;
+  type Source = fields.SchemaField.InnerPersistedType<Schema>;
+
+  interface Schema extends DataSchema {
+    /**
+     * The _id which uniquely identifies this TableResult embedded document
+     * @defaultValue `null`
+     */
+    _id: fields.DocumentIdField;
+
+    /**
+     * A result subtype from CONST.TABLE_RESULT_TYPES
+     * @defaultValue `CONST.TABLE_RESULT_TYPES.TEXT`
+     */
+    type: fields.NumberField<{
+      required: true;
+      choices: CONST.TABLE_RESULT_TYPES[];
+      initial: typeof CONST.TABLE_RESULT_TYPES.TEXT;
+      validationError: "must be a value in CONST.TABLE_RESULT_TYPES";
+    }>;
+
+    /**
+     * The text which describes the table result
+     * @defaultValue see {@link fields.HTMLField}
+     */
+    text: fields.HTMLField;
+
+    /**
+     * An image file url that represents the table result
+     * @defaultValue `null`
+     */
+    img: fields.FilePathField<{ categories: ["IMAGE"] }>;
+
+    /**
+     * A named collection from which this result is drawn
+     * @defaultValue `""`
+     */
+    documentCollection: fields.StringField;
+
+    /**
+     * The _id of a Document within the collection this result references
+     * @defaultValue `null`
+     */
+    documentId: fields.ForeignDocumentField<Document, { idOnly: true }>;
+
+    /**
+     * The probabilistic weight of this result relative to other results
+     * @defaultValue `1`
```

Comment:
I'm not sure, but I think this should be `null`.

Reply:
Confirmed. Updated.